src/kotlin/org/intellij/plugins/hcl/terraform/config/model/TypeModelLoader.kt

```diff
@@ -347,7 +348,24 @@ class TypeModelLoader(val external: Map<String, TypeModelProvider.Additional>) {
     val conflicts: List<String> = value.array<String>("ConflictsWith")?.map { it } ?: emptyList()
 
     val deprecated = value.string("Deprecated")
-    val has_default: Boolean = value.obj("Default")?.isNotEmpty() ?: false
+    val defaultObj = value.obj("Default")
+    val defaultValue = if (defaultObj == null || defaultObj.isEmpty()) {
+      null
+    }
+    else {
+      val defaultValueString = defaultObj.string("Value")
+      when (typeString) {
+        null -> null
+        "Bool", "TypeBool" -> defaultValueString?.toBoolean() ?: false
+        "Int", "TypeInt" -> defaultValueString?.toInt() ?: 0
+        "Float", "TypeFloat" -> defaultValueString ?: "0"
+        "String", "TypeString" -> defaultValueString ?: ""
+        else -> {
+          LOG.warn("Unhandled default type $typeString for $fqn")
```

Comment:
`logErrorAndFailInInternalMode` could be used where to detect unsupported types in tests